qubes/app.py

```diff
@@ -1424,6 +1425,20 @@ def register_event_handlers(self, old_connection=None):
                 self._domain_event_callback,
                 None))
 
+        # Stop storage for domains not currently running
+        loop = asyncio.get_event_loop()
+        async def stop_storage():
+            future = (qubes.utils.void_coros_maybe(
+                i.storage.stop() for i in self.domains if not i.is_running()),)
```

Comment:
This should take `i.startup_lock`, to avoid starting a VM when `storage.stop()` is still running.
(even if you think it couldn't happen for other reasons, better to take the lock anyway - see an example above why such assumptions are dangerous...)